aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1896,6 +1911,29 @@ module aptos_framework::stake {
         join_validator_set(&validator, validator_address);
     }
 
+    #[test]
+    public entry fun test_rewards_calculation() {
```

Comment:
can you add a test that would fail before, for overflow?

Reply:
Good point. Just added a test case for this.